src/misc.cpp

```diff
@@ -149,11 +146,14 @@ argtokey_callback_t argToKey(const Local<Value> &val, MDB_val &key, NodeLmdbKeyT
         }
         
         isValid = true;
-        currentUint32Key = val->Uint32Value();
+        uint32_t* uint32Key = new uint32_t;
```

Comment:
Is the dynamic allocation really necessary here?

Reply:
I think so. We can't rely on stack allocation since this key is used outside the stack. And you had previously gotten around dynamic allocation by using a thread_local, and using a single slot that would be used sequentially in a single thread. But this key is now potentially going to be used across different threads, so a thread_local will no longer work, and I think we really do need to allocate memory for it until it is actually used (and we could have multiple keys allocated at once). Anyway, maybe I am missing something, and there is another way?